docs/source/examples/e2e_flow.rst

```diff
@@ -0,0 +1,438 @@
+
+    [_checkpointer.py:473] Model checkpoint of size 9.98 GB saved to <checkpoint_dir>/hf_model_0001_0.pt
+
+    [_checkpointer.py:473] Model checkpoint of size 3.50 GB saved to <checkpoint_dir>/hf_model_0002_0.pt
+
+    [_checkpointer.py:484] Adapter checkpoint of size 0.01 GB saved to <checkpoint_dir>/adapter_0.pt
+
+
+The "merged weights" (see the :ref:`LoRA Tutorial <lora_finetune_label>` for more details)
+are split across two checkpoint files similar to the source checkpoints from the HF Hub.
+In fact the keys would be identical between these checkpoints. For more details see the
+checkpointing tutorial. We also have a third checkpoint file which is much smaller in size
+and contains the learnt LoRA adapter weights. For this tutorial, we'll only use the model
+checkpoints and not the adapter weights.
+
+|
+
+Run Evaluation using EleutherAI's Eval Harness
+----------------------------------------------
+
+We've fine-tuned a model. But how well does this model really do? Let's run some Evaluations!
+
+Evaluation is a hard problem. Instead of re-inventing the wheel, TorchTune integrates with
+EleutherAI's evaluation harness. An example of this is available through the
+``eleuther_eval`` recipe. In this tutorial, we're going to directly use this recipe by
+modifying it's associated config ``eleuther_eval.yaml``.
+
+Let's first copy over the config to our local working director so we can make changes.
+
+.. code-block:: bash
+
+    tune cp eleuther_eval ./custom_eval_config.yaml
+
+Let's modify ``custom_eval_config.yaml`` to include the following changes.
+
+.. code-block:: yaml
+
+    checkpointer:
+        _component_: torchtune.utils.FullModelHFCheckpointer
+
+        # directory with the checkpoint files
+        # this should match the output_dir specified during
+        # finetuning
+        checkpoint_dir: <checkpoint_dir>
+
+        # checkpoint files for the fine-tuned model. This should
+        # match what's shown in the logs above
+        checkpoint_files: [
+            hf_model_0001_0.pt,
+            hf_model_0002_0.pt,
+        ]
+
+        output_dir: <checkpoint_dir>
+        model_type: LLAMA2
+
+    # Make sure to update the tokenizer path to the right
+    # checkpoint directory as well
+    tokenizer:
+        _component_: torchtune.models.llama2.llama2_tokenizer
+        path: <checkpoint_dir>/tokenizer.model
+
+
+Once the config is updated, let's kick off evaluation! We'll use the
+``truthfulqa_mc2`` task which is also the default in the config.
+
+.. code-block:: bash
+
+    tune run eleuther_eval --config ./custom_eval_config.yaml
+
+
+Once evaluation is complete, you'll see the following in the logs.
+
+.. code-block:: bash
+
+    [evaluator.py:324] Running loglikelihood requests
+    [eleuther_eval.py:195] Eval completed in 121.27 seconds.
+    [eleuther_eval.py:197] truthfulqa_mc2: {'acc,none': 0.48919958543950917 ...}
+
+So seems like our fine-tuned model gets ~48% on this task. Which is pretty good.
```

Comment:
```suggestion
So seems like our fine-tuned model gets ~48% on this task, which is pretty good.
```

Reply:
I feel like this line is awkward. Why not just combine with the subsequent sentences plus a bit describing what truthfulqa_mc2 is actually doing? Basically something like:


The Truthful QA dataset measures a model's propensity to be truthful when answering questions. Specifically, we will evaluate our model on the truthfulqa_mc2 task, which measures the model's zero-shot accuracy on a question followed by one or more true responses and one or more false responses. We can run evaluation on our downloaded checkpoints first as a baseline
... (hopefully the command should just be a config change to the directory)

Now, we evaluate our fine-tuned model 
...

We can see that the fine-tuned model yields an 8% overall improvement in zero-shot accuracy on this task.

Reply:
+1 to Evan's paragraphs, much more compelling and informative

Reply:
Updated!